src/widgets/block.rs

```diff
@@ -532,6 +533,16 @@ impl<'a> Block<'a> {
         inner
     }
 
+    fn have_title_at_position(&self, position: Position) -> bool {
+        if self.titles.iter().any(|t| t.position == Some(position)) {
+            return true;
+        }
+        if self.titles.iter().any(|t| t.position.is_none()) {
+            return self.titles_position == position;
+        }
+        false
```

Comment:
slightly simpler (nit):

```suggestion
        self.titles
            .iter()
            .any(|title|
                title.position == Some(position)
                || title.position.is_none() && self.titles_position == position)
```

Reply:
How about:

``` rust
        self.titles.iter().any(|title| match title.position {
            Some(title_position) => title_position == position,
            None => self.titles_position == position,
        })
```

Reply:
Good idea, but you could go one step futher still: :)
```rust
self.titles.iter.any(|title| position == title_position.unwrap_or(self.titles_position))
```